packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -188,134 +243,78 @@ class ButtonBase extends React.Component {
         onClick(event);
       }
     }
-  };
-
-  handleKeyUp = event => {
-    if (this.props.focusRipple && event.key === ' ' && this.ripple && this.state.focusVisible) {
-      this.keyDown = false;
+  });
+  const handleKeyUp = useEventCallback(event => {
+    const { current: ripple } = rippleRef;
+    if (focusRipple && event.key === ' ' && ripple && focusVisible) {
+      keydownRef.current = false;
       event.persist();
-      this.ripple.stop(event, () => {
-        this.ripple.pulsate(event);
+      ripple.stop(event, () => {
+        rippleRef.current.pulsate(event);
```

Comment:
It's strange, we do
```js
const { current: ripple } = rippleRef;
ripple.x
```

and
```js
rippleRef.current.x
```

at the same time. Can we consolidate it around a single pattern?

Reply:
Agreed. I went with destructuring only until I realized that I would access a possibly stale value in the callback.

Reply:
Addressed